webapp/javascript/components/TimelineChart/ContextMenu.plugin.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import * as ReactDOM from 'react-dom';
+import { randomId } from '@webapp/util/randomId';
+import { Provider } from 'react-redux';
+import store from '@webapp/redux/store';
+
+// Pre calculated once
+// TODO(eh-am): does this work with multiple contextMenus?
+const WRAPPER_ID = randomId('contextMenu');
+
+export interface ContextMenuProps {
+  click: {
+    /** The X position in the window where the click originated */
+    pageX: number;
+    /** The Y position in the window where the click originated */
+    pageY: number;
+  };
+  timestamp: number;
+}
+
+(function ($: JQueryStatic) {
+  let globalPlot: jquery.flot.plot;
+
+  function onClick(
+    event: unknown,
+    pos: { x: number; pageX: number; pageY: number }
+  ) {
+    const container = inject($);
+    const containerEl = container?.[0];
+
+    // unmount any previous menus
+    ReactDOM.unmountComponentAtNode(containerEl);
+
+    // TODO(eh-am): improve typing
+    const ContextMenu = (globalPlot.getOptions() as ShamefulAny).ContextMenu as
+      | React.FC<ContextMenuProps>
+      | undefined;
+
+    if (ContextMenu) {
+      // TODO(eh-am): why do we need this conversion?
+      const timestamp = Math.round(pos.x / 1000);
+
+      // Add a Provider (reux) so that we can communicate with the main app via actions
+      // idea from https://stackoverflow.com/questions/52660770/how-to-communicate-reactdom-render-with-other-reactdom-render
+      // TODO(eh-am): add a global Context too?
+      ReactDOM.render(
+        <Provider store={store}>
+          <ContextMenu click={{ ...pos }} timestamp={timestamp} />
+        </Provider>,
+        containerEl
+      );
+    }
+  }
+
+  function init(plot: jquery.flot.plot & jquery.flot.plotOptions) {
+    // update closure
+    globalPlot = plot;
```

Comment:
added a comment

Reply:
>// update closure so that onClick can access the plot instance
you have access to plot instance without this data mutation, you can just move `onClick` f-n under init(plot: PlotType) function, in this case you can access plot from upper scope.

Reply:
good point, fixed it!